### PR TITLE
Boxfish.ts supports new HPK  upgrader

### PR DESCRIPTION
**Pull request by jontore**

### CODE REVIEW

File: src/components/device/boxfish.ts [hunk 1]
```diff
@@ -110,6 +110,7 @@ export default class Boxfish extends UvcBaseDevice implements IDeviceManager {
         resolve();
       });
       upgrader.once(CameraEvents.UPGRADE_FAILED, (reason) => {
+        console.log('UPGRADE FAILED');
```
Comment (brikendr) [inline src/components/device/boxfish.ts]: Use the logger instead of doing console.log

File: src/components/upgrader/boxfishUpgraderFactory.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+const HPK_SUPPORT_VERSION = '1.1.0-0';
+export default async function createBoxfishUpgrader(
+  manager: IDeviceManager,
+  sdkDeviceDiscoveryEmitter: EventEmitter,
+  logger: any
+): Promise<IDeviceUpgrader> {
+  const info = await manager.getInfo();
+  if (semver.gte(info.version, HPK_SUPPORT_VERSION)) {
+    return new HPKUpgrader(manager, sdkDeviceDiscoveryEmitter, logger);
```
Comment (brikendr) [inline src/components/upgrader/boxfishUpgraderFactory.ts]: Nice

File: src/components/upgrader/hpkUpgrader.ts [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+          }
+          if (statusMessage.error_count > 0) {
+            return reject(statusMessage);
+          }
+          const elapsedPoints = statusMessage.elapsed_points || 0;
+          const progressPercentage = (elapsedPoints / totalProgressPoints) * 100;
+          this._logger.info(`Upgrading HPK: Status: ${Math.ceil(progressPercentage)}% step ${statusMessage.operation}\r`);
+          this.emit(CameraEvents.UPGRADE_PROGRESS, {
+            operation: statusMessage.operation,
+            progress: progressPercentage
```
Comment (brikendr) [inline src/components/upgrader/hpkUpgrader.ts]: Awesome!

File: src/components/upgrader/hpkUpgrader.ts [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        await this.doUpgrade();
+      } catch (e) {
+        this._logger.error('Upgrade failed', e);
+        this.emit(CameraEvents.UPGRADE_FAILED, e);
+        throw e;
+      }
+
+      this.once('UPGRADE_REBOOT_COMPLETE', async () => {
+        try {
+          await this.doUpgrade();
```
[the remark this answers is not in the document]
Comment (jontore) [reply src/components/upgrader/hpkUpgrader.ts]: First time it upgrades, second time it verifies the installation.

**Pull request merged**